### PR TITLE
[ADD] Create BR from issues

### PR DESCRIPTION
**Pull request by rubencabrera**
Create a BR from issue as discussed in #171 

Adds a button which opens a wizard, taking default info from the issue to fill the BR. 
The correspondence of fields from issue to BR is:
Name -> Description
Priority
Description -> Business Requirement ("user story")
Assigned to (user_id) -> Requested id
Partner id
Assigned to (user_id) -> Responsible id
Project id

To be reviewed is set to True by default
Assigned to (user_id) -> Reviewer ids

### CODE REVIEW

File: business_requirement_from_support/models/project_issue.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8 -*-
+# © 2017 Praxya (https://www.praxya.com).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openerp import api, fields, models, _
+
+
+class ProjectIssue(models.Model):
+    _inherit = "project.issue"
+
+    business_requirement_id = fields.Many2one(
```
Comment (elicoidal) [inline business_requirement_from_support/models/project_issue.py:10]: Dont you want to add this link (read only or not) in the issue form view?

File: business_requirement_from_support/README.rst [hunk 2]
```diff
@@ -0,0 +1,73 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: https://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+====================================
+Business Requirement Deliverable CRM
```
Comment (elicoidal) [inline business_requirement_from_support/README.rst]: Business Requirement from Issues

File: business_requirement_from_support/README.rst [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+====================================
+Introduction
+============
+
+This module is part of a set of modules 
+(`Business Requirements <https://github.com/OCA/business-requirement/blob/8.0/README.md>`_) 
+
+This module adds integration between project issues  and Business Requirement
```
Comment (elicoidal) [inline business_requirement_from_support/README.rst]: remove extra space

File: business_requirement_from_support/__openerp__.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    'name': 'Business Requirement from Support',
+    'category': 'Business Requirements Management',
+    'summary': 'Generate Business Requirements from issues',
+    'version': '8.0.1.0.0',
+    'website': 'https://www.praxya.com/',
+    "author": "Rubén Cabrera Martínez",
+    'depends': [
+        'business_requirement_deliverable_project',
```
Comment (elicoidal) [inline business_requirement_from_support/__openerp__.py]: Do you need this one or only business_requirement?

File: business_requirement_from_support/models/project_issue.py (hunk 1 above)
Comment (elicoidal) [reply business_requirement_from_support/models/project_issue.py:10]: Side note: what if the button is clicked 2 times? What would be the expected behavior?

**Review by elicoidal — CHANGES_REQUESTED** [review]
:+1: Thanks for the contribution!

File: business_requirement_from_support/models/project_issue.py (hunk 1 above)
Comment (rubencabrera) [reply business_requirement_from_support/models/project_issue.py:10]: The answer to the first one is yes, I just forgot, so I guess the button should be hidden if the issue has a related BR already, do you agree?

File: business_requirement_from_support/__openerp__.py (hunk 4 above)
Comment (rubencabrera) [reply business_requirement_from_support/__openerp__.py]: It made sense to include this one as it is the next logical step to take, generate tasks, but you're right, it is not a dependency, I'll change it to business_requirement.

**Comment by elicoidal** [comment]
@rubencabrera Do you have any update on this PR?

**Comment by rubencabrera** [comment]
Is there anything else to add? On the last 4 commits I made all the changes you requested and fixed the Travis.

**Comment by elicoidal** [comment]
OK: I will test it and come back to you :)

**Comment by elicoidal** [comment]
By the way: do not hesitate to ping me when a new version is ready as I do not necessarily closely follow all PR/modifs @rubencabrera

**Review by elicoidal — APPROVED** [review]
@rubencabrera tested it and works like a charm.

Do you think we should add a smart button in the BR to list the related issues?

**Comment by elicoidal** [comment]
@victormartinelicocorp @seb-elico @dreispt

File: business_requirement_from_support/views/project_issue.xml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+<openerp>
+    <data>
+        <record id="view_create_br_form" model="ir.ui.view">
+            <field name="name">view.issue.form</field>
+            <field name="model">project.issue</field>
+            <field name='inherit_id' ref='project_issue.project_issue_form_view'/>
+            <field name="arch" type="xml">
+                <xpath expr="//field[@name='stage_id']" position="before">
+                    <button string="Create BR"
+                            attrs="{'invisible':[('business_requirement_id', '!=', False)]}"
```
Comment (seb-elico) [inline business_requirement_from_support/views/project_issue.xml]: I would add a space after `:` to improve readability

File: business_requirement_from_support/wizard/br_from_issue_view.xml [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+                <group >
+                    <field name="name" required="1"/>
+                    <field name="requested_id" required="1"/>
+                    <field name="responsible_id" required="1"/>
+                    <field name="partner_id" required="1"/>
+                    <field name="project_id" required="1"/>
+                    <field name="to_be_reviewed"/>
+                    <field name="priority" widget="priority"/>
+                    <field name="reviewer_ids" widget="many2many_tags"
+                           attrs="{'invisible':[('to_be_reviewed', '!=', True)]}"
```
Comment (seb-elico) [inline business_requirement_from_support/wizard/br_from_issue_view.xml]: I would add a space after `:` to improve readability

File: business_requirement_from_support/wizard/br_from_issue_view.xml [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+          </field>
+       </record>
+       <!-- crm make sale's  action -->
+       <record id="action_create_br" model="ir.actions.act_window">
+          <field name="name">Business Requirement from issue</field>
+          <field name="type">ir.actions.act_window</field>
+          <field name="res_model">br.issue</field>
+          <field name="view_type">form</field>
+          <field name="view_mode">form</field>
+          <!--<field name="view_id" ref="create_br_view" />-->
```
Comment (seb-elico) [inline business_requirement_from_support/wizard/br_from_issue_view.xml]: Dead code?

File: business_requirement_from_support/wizard/br_from_issue_view.xml [hunk 8]
```diff
@@ -0,0 +1,41 @@
+<openerp>
+   <data>
+       <!-- Make BR from issue view -->
+       <record id="view_create_br_from_issue" model="ir.ui.view">
+          <field name="name">create.br.form</field>
+          <field name="model">br.issue</field>
+          <field name="arch" type="xml">
+            <form string="Business Requirement from Issue" >
```
Comment (seb-elico) [inline business_requirement_from_support/wizard/br_from_issue_view.xml]: Tag closing is not consistent (sometimes there's a space before `>` or `/>`, sometimes not)

**Review by seb-elico — CHANGES_REQUESTED** [review]
Few cosmetic changes, overall LGTM

File: business_requirement_from_support/wizard/br_from_issue_view.xml (hunk 8 above)
Comment (rubencabrera) [reply business_requirement_from_support/wizard/br_from_issue_view.xml]: That's a dilemma I always face, is there any convention on that matter? I'll take whatever you say as I have no opinion on this.

Comment (elicoidal) [reply business_requirement_from_support/wizard/br_from_issue_view.xml]: lol no space is good ;)

**Comment by rubencabrera** [comment]
I like the smart button with related issues on the BR, will add it ASAP (hope it's a few hours).

**Comment by elicoidal** [comment]
@rubencabrera Any feedback on the smart button?

**Comment by elicoidal** [comment]
@rubencabrera any news?

**Comment by rubencabrera** [comment]
Working on it right now, @elicoidal, sorry for the delay, been on holidays and came back to a ton of issues.

**Comment by elicoidal** [comment]
hehe good to have you back @rubencabrera 
A lot of new features and bug fixing have landed in the repo while you were away.
Demo data is almost ready and major refactoring in the BR/project link is successful (although tricky for the migration..).
If everything is fine we start migration to v10 next week

**Comment by rubencabrera** [comment]
Merged main branch commits and added smart button for issues. The tree view might need some changes, please comment any suggestions.

**Review by elicoidal — APPROVED** [review]
We need to make sure that this works after #192 is merged (but I do not see why it shouldnot)

**Comment by elicoidal** [comment]
@victormartinelicocorp

**Comment by elicoidal** [comment]
@dreispt @pedrobaeza Can you help?

**Comment by rubencabrera** [comment]
If  #192 is ready or almost ready, I can check mergeability or even do a PR to their contributions with what has been committed here. Whatever makes it easier to speed this PR up.

**Comment by elicoidal** [comment]
> If #192 is ready or almost ready, I can check mergeability or even do a PR to their contributions with what has been committed here. Whatever makes it easier to speed this PR up.

It is: we are now finalizing with details on how to show the data. Model should not change.
Review / tests are welcome :)

**Review by martintorresvm — APPROVED** [review]

**Comment by elicoidal** [comment]
@rubencabrera #192 has been merged: you can rebase and test.

**Comment by elicoidal** [comment]
@rubencabrera your PR looks strange: too many modifications on modules that you should not impact.
Can you recheck?

**Comment by rubencabrera** [comment]
I did a git fetch orign 8.0 into my 8.0 branch and then a git rebase origin/8.0 on the PR branch. The modifications are from the commits in 8.0.

**Comment by elicoidal** [comment]
@rubencabrera I still think something is wrong. Can you recheck? many of the changes proposed by the PR are already incorporated in the core now.

**Comment by rubencabrera** [comment]
Ok, @elicoidal, looks fixed now. It seems I forgot the --rebase flag.

**Review by elicoidal — APPROVED** [review]

**Comment by elicoidal** [comment]
Good to go!

**Pull request merged**